src/Widgets/DiskBar.vala

```diff
@@ -0,0 +1,85 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2016-2018 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Michael Aaron Murphy <michael@system76.com>
+ */
+
+public class Installer.DiskBar: Gtk.Box {
+    public string disk_name;
+    public string disk_path;
+    public uint64 size;
+
+    public Gtk.Box label;
+
+    public DiskBar (
+        string model,
+        string path,
+        uint64 size,
+        GLib.Array<PartitionBar> partitions
+    ) {
+        this.disk_name = model;
```

Comment:
GObject style please :)

Reply:
I don't quite get the point of the `Object` thing in Vala. Some input parameters are incompatible with it, so you end up having an odd mix of Object + non-Object initialization.